core/src/main/java/com/zextras/carbonio/files/rest/services/ProcedureService.java

```diff
@@ -0,0 +1,131 @@
+      + properties.getProperty(Mailbox.URL, "127.78.0.2")
+      + ":"
+      + properties.getProperty(Mailbox.PORT, "20004");
+    storagesUrl = "http://"
+      + properties.getProperty(Storages.URL, "127.78.0.2")
+      + ":"
+      + properties.getProperty(Storages.PORT, "20002")
+      + "/";
+  }
+
+  /**
+   * Allows to upload the {@param nodeId} to the carbonio mailbox store. These are the target module
+   * supported:
+   * <ul>
+   *   <li>{@link TargetModule#MAILS}</li>
+   *   <li>{@link TargetModule#CALENDARS}</li>
+   *   <li>{@link TargetModule#CONTACTS}</li>
+   * </ul>
+   *
+   * @param nodeId is a {@link String} of the node id to upload.
+   * @param targetModule is a {@link TargetModule} representing the module to upload.
+   * @param requester is a {@link User} representing the requester of this operation.
+   * @param cookiesRequester is a {@link String} representing the requester cookies necessary to
+   * perform the upload operation.
+   *
+   * @return a {@link Try} containing a {@link String} representing the mailbox attachment id
+   * uploaded or a {@link Throwable} if something goes wrong. These are the possible failures:
+   * <ul>
+   *   <li>{@link BadRequestException} when the node is not a file</li>
+   *   <li>{@link Exception} when the download of the blob to upload fails</li>
+   *   <li>{@link Exception} when the {@link TargetModule} is {@link TargetModule#CHATS} because,
+   *   for now, it is not supported</li>
+   * </ul>
+   */
+  public Try<String> uploadToModule(
+    UUID nodeId,
+    TargetModule targetModule,
+    User requester,
+    String cookiesRequester
+  ) {
+
+    if (!targetModule.equals(TargetModule.CHATS)) {
+      Node nodeToUpload = nodeRepository.getNode(nodeId.toString()).get();
+
+      if (!nodeToUpload.getNodeType().equals(NodeType.FOLDER)) {
+        FileVersion fileVersion = fileVersionRepository.getLastFileVersion(nodeId.toString()).get();
+        InputStream blob;
+        try {
+          blob = StoragesClient
+            .atUrl(storagesUrl)
+            .download(FilesIdentifier.of(
+              nodeId.toString(),
+              nodeToUpload.getCurrentVersion(),
+              requester.getUuid())
+            );
+        } catch (Exception exception) {
+          logger.error(MessageFormat.format("Failed to download the node: {0}", nodeId));
+          return Try.failure(new InternalServerErrorException(exception));
+        }
+
+        return MailboxHttpClient
+          .atURL(mailboxUrl)
+          .uploadFile(
+            cookiesRequester,
+            nodeToUpload.getFullName(),
+            fileVersion.getMimeType(),
+            blob
+          );
+      }
+      logger.error(MessageFormat.format("Folder cannot be uploaded to {0} store", targetModule));
+      return Try.failure(new BadRequestException());
+    }
+
+    return Try.failure(
+      new InternalServerErrorException(MessageFormat.format("{0} not supported", TargetModule.CHATS))
```

Comment:
Is this code linted? It seems too long

Reply:
Nice catch! The line was 121 characters long! I fixed it